### ISSUE TITLE
Create generative model

### ISSUE THREAD

**Issue by david-thrower** [david-thrower/cerebros-core-algorithm-alpha#146]
Based on the provided information, here's a step-by-step guide to address the issues and enhance the model:

1. Create a new branch from the specified commit:
2. Duplicate the GPT-2 based phishing classification notebook for this project.
3. Modify the final layer of the model:

Replace the `Dense(1)` layer with a `Dense(VOCAB_SIZE)` layer and change the loss function to `sparse_categorical_crossentropy` (since you're using a softmax activation function).
4. Handle de-tokenization:

To handle de-tokenization without affecting the loss, you can create a custom function that takes the one-hot encoded output and returns the corresponding word. This function should ignore the loss calculation during training.
```python
def de_tokenize(y_pred):
    return np.argmax(y_pred, axis=-1)

model.compile(loss="sparse_categorical_crossentropy", optimizer="adam", metrics=["accuracy"])
```
5. Prepare the training dataset:

6. Create a function that takes a prompt and an expected response and generates a list of text sequences where each has one additional word cumulatively added from the expected response.:
  1. E.g. the data set `{"prompt":"Write a haiku about integrity: ", "response":["Integrity is great\nIntegrity is nice.\nIntegrity will get you far" }` will become these data and labels: `data = ["Write a haiku about integrity:", "Write a haiku about integrity: Integrity", "Write a haiku about integrity: Integrity will" ]`; `labels = ["Integrity", "will", "get"]`
6. Create a benchmark training set: like the above.
8. Prototype a benchmark training loop.

**Comment by david-thrower** [comment]
A prototype of packaging the text samples:

```python3
import numpy as np


# Text Sample: Class for a text sample

class TextSample:
    def __init__(self, prompt: str, response: str):
        self.prompt = prompt
        self.response = response


sample1 = TextSample(prompt="Tell me all about the capitol of France",
                     response="Paris is known as the city of love")
sample2 = TextSample(prompt="Write a haiku about life",
                     response="Life blows.\nYou go to school.\nYou go to work")
sample3 = TextSample(prompt="Write an ode to Scilence:",
                     response="Silence is awesome. Silence is rare. Silence is beauty. Silence is nowhere.")

samples = [sample1, sample2, sample3]


data = []
labels = []


def create_data_and_labels(samples):
    for i in np.arange(len(samples)):
        sample_0 = samples[i]
        prompt_0_str = sample_0.prompt
        response_0_str = sample_0.response

        response_0_list = response_0_str.replace(
            "\n", " \n ").replace(".", " . ").split(" ")

        data_0 = []
        label_0 = []
        data_0 = prompt_0_str
        for j in np.arange(len(response_0_list)):
            if i == 0:
                label_0 = response_0_list[j]
            else:
                data_0 += f" {response_0_list[j - 1]}"
                label_0 = response_0_list[j]
            data.append(data_0)
            labels.append(label_0)


create_data_and_labels(samples)

for data_0, label_0, in zip(data, labels):
    print("Sample:")
    print(data_0)
    print(label_0)
```
